### ISSUE TITLE
prefer-arrow-callback rule parameter for "never"

### ISSUE THREAD

**Issue by shellscape** [eslint/eslint#9302]
Following #9301, this rule could also benefit from a "never" parameter for completeness, similar to the "never" parameter for https://eslint.org/docs/rules/object-shorthand. 

**Tell us about your environment**

* **ESLint Version:** 4.4.1
* **Node Version:** 8.2.1
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
We're using this and extending it https://github.com/webpack-contrib/eslint-config-webpack

<details>
<summary>Configuration</summary>

```js
{
  "extends": "webpack"
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
function thing (cb) { cb(); }

thing(function () { });
```

transformed to:

```js
function thing (cb) { cb(); }

thing(() => {});
```

`eslint-config-webpak` contains `"prefer-arrow-callback": "error"` Nothing out of the ordinary there. In the project I'm working on, we have a few files which are served to a client (browser) that we are not transpiling, which don't yet support arrow functions.

**What did you expect to happen?**

Well we expected that behavior.

**What actually happened? Please include the actual, raw output from ESLint.**

Something along the lines of `"prefer-arrow-callback": ["error", "never"]` would be immensely useful for preventing devs from writing code using arrow functions for ES5 environments. Additionally setting the `"ecmaVersion": 5` isn't an option since it doesn't complain about arrow functions (it does however, pick up other things like `const`).

**Comment by not-an-aardvark** [comment]
Thanks for the proposal.

> Additionally setting the `"ecmaVersion": 5` isn't an option since it doesn't complain about arrow functions 

I don't think this is correct when using the default parser (other parsers sometimes handle `parserOptions` differently).

If you want to disallow arrow functions entirely, I think it would be better to use the [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax) rule -- that way, you can disallow arrow functions everywhere, rather than just in callbacks.

**Comment by shellscape** [comment]
For completeness, would this not be a reasonable addition to the rule?

I think for the time being I could get by with the no-restrcited-syntax rule, however. Appreciate the suggestion.

**Comment by platinumazure** [comment]
I like the idea of adding an option for autofix capability, but I agree that the value is only for autofix and that `no-restricted-syntax` should be sufficient for most cases.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.

**Comment by shellscape** [comment]
@not-an-aardvark I'd be curious to know if the entire team was directed to put eyes on this before it was closed. (yay for canned responses)

**Comment by ilyavolodin** [comment]
@shellscape We can't force the whole team to look at every issue. Most of the team go by issue notifications and look over most (if not all) of the new issues. If they didn't respond to the issue, it usually means they didn't feel that it was something they would want to support. We found that the longer the issue stays open without activity, the less of a chance there is for support from the team and implementation, that's why we choose to close issues after a period of time.
